libdevice/crt_wrapper.cpp

```diff
@@ -9,6 +9,14 @@
 #include "wrapper.h"
 
 #ifdef __SPIR__
+DEVICE_EXTERN_C
+div_t div(int x, int y) { return {x / y, x % y}; }
```

Comment:
We have to allow device runtimes to provide their optimized versions of these functions, so we have to call `__devicelib_` variants here, and put the actual implementation into the fallback part of the library.

Reply:
> We have to allow device runtimes to provide their optimized versions of these functions, so we have to call `__devicelib_` variants here, and put the actual implementation into the fallback part of the library
To some complicated functions such as __divsc3/__mulsc3, putting the actual implementation into fallback spv is valuable since device runtime may have interest providing a more optimized version. To div, it looks like the implementation is too simple, I doubt whether device runtime would like to provide a more optimized version. If device runtime wouldn't like or there is not much value to optimize such "simple" functions, just providing the actual implementation in wrapper obj may be conciser.
Thanks very much.

Reply:
> > We have to allow device runtimes to provide their optimized versions of these functions, so we have to call `__devicelib_` variants here, and put the actual implementation into the fallback part of the library
> > To some complicated functions such as __divsc3/__mulsc3, putting the actual implementation into fallback spv is valuable since device runtime may have interest providing a more optimized version. To div, it looks like the implementation is too simple, I doubt whether device runtime would like to provide a more optimized version. If device runtime wouldn't like or there is not much value to optimize such "simple" functions, just providing the actual implementation in wrapper obj may be conciser.
> > Thanks very much.

I think there may be a hardware instruction that computes both values on some devices, so it does make sense to have devicelib api for these operations.

Reply:
> > > We have to allow device runtimes to provide their optimized versions of these functions, so we have to call `__devicelib_` variants here, and put the actual implementation into the fallback part of the library
> > > To some complicated functions such as __divsc3/__mulsc3, putting the actual implementation into fallback spv is valuable since device runtime may have interest providing a more optimized version. To div, it looks like the implementation is too simple, I doubt whether device runtime would like to provide a more optimized version. If device runtime wouldn't like or there is not much value to optimize such "simple" functions, just providing the actual implementation in wrapper obj may be conciser.
> > > Thanks very much.
> 
> I think there may be a hardware instruction that computes both values on some devices, so it does make sense to have devicelib api for these operations.

Hi, @vzakhari 
If so, moving div into fallback spv makes sense. We have another libc item to add into devicelib which is also very simple: 
int abs(int x);
Do you think we need to move its actual implementation into fallback lib or just provide the actual impl in wrapper obj?
I think the impl is very simple:
int abs(int x) {  return i < 0 ? -i : i; }

Thanks very much.

Reply:
I think it is always easier to follow the common pattern/design, than to introduce exceptions :)